server/v2/stf/stf.go

```diff
@@ -153,7 +152,7 @@ func (s STF[T]) deliverTx(
 	ctx context.Context,
 	state store.WriterMap,
 	tx T,
-	execMode corecontext.ExecMode,
+	execMode transaction.ExecMode,
```

Comment:
This however doesn't contain all exec mode, just exec mode available for modules. So imho it should live in stf directly

Reply:
The proposal then would be to use an internal exec modes? I think it’s fine, they can be packed as uint8 into context and the service can cast so that the exec mode can properly transit the boundary to modules.

Reply:
Yes, baseapp and stf have their exec mode, and modules use the transaction service

Reply:
Are you going to do this in this pR btw?

Reply:
Yea working on it today

Reply:
does this need to change to internal?

Reply:
done!